hyperjumptech/monika
# Restructure PING request

# Monika Pull Request (PR)

## What feature/issue does this PR add
1. Implements ticket #1092 
2. Ping/icmp request piggy backed onto the http request probe
3. Redo the ping/icmp probe as a stand alone probe

## How did you implement / how did you fix it
1.  Add specifing PingProber
2. Add entry to json schema for ping/icmp requests


## How to test
1. Use the following test yaml: 
```yaml
probes:
  - id: 'new-ping'
    name: 'ping test'
    ping:
      - uri: https://hyperjump.tech

  - id: 'old-ping'
    name: 'status-200-test'
    requests:
      - url: https://google.com
        ping: true

notifications:
  - id: desktop
    type: desktop
```
2. Run monika with the above config `npm start -- -c config.yaml -s -1`
3. Both PING requests should work
4. The general `npm run test` should all pass

## Screenshots

Valid ping hosts (old and new format)
![image](https://github.com/hyperjumptech/monika/assets/964106/e57cc194-6d94-4c95-a6dc-63f2b0022e7e)

Invalid host (single host)
![image](https://github.com/hyperjumptech/monika/assets/964106/83702f75-2555-42d8-aa27-c2abd449b218)

Invalid config.yaml (adding bad field to ping)

Integrated Json schema validator on IDE (vscode example)
![image](https://github.com/hyperjumptech/monika/assets/964106/50d2ec64-7f6e-4603-9160-a5e98c66b175)

Console error on bad config
![image](https://github.com/hyperjumptech/monika/assets/964106/96d4dd76-01e6-4c09-9dc9-4ff43bb3b0a5)
